src/platform/forms-system/src/js/helpers.js

```diff
@@ -246,47 +246,63 @@ export function filterInactivePageData(inactivePages, activePages, form) {
   );
 }
 
-export function stringifyFormReplacer(key, value) {
-  // an object with country is an address
-  if (
-    value &&
-    typeof value.country !== 'undefined' &&
-    (!value.street || !value.city || (!value.postalCode && !value.zipcode))
-  ) {
-    return undefined;
-  }
+function isPartialAddress(data) {
+  return (
+    data &&
+    typeof data.country !== 'undefined' &&
+    (!data.street || !data.city || (!data.postalCode && !data.zipcode))
+  );
+}
 
-  // clean up empty objects, which we have no reason to send
-  if (typeof value === 'object') {
-    const fields = Object.keys(value);
-    if (
-      fields.length === 0 ||
-      fields.every(field => value[field] === undefined)
-    ) {
+/**
+ * Create a replacer function for JSON.stringify
+ * A replacer function is the second argument to JSON.stringify
+ *
+ * @param {{
+ *  allowPartialAddress?: boolean,
+ * }} [options] - Options for the replacer
+ */
+export function createStringifyFormReplacer(options) {
+  const replacerFn = (key, value) => {
+    if (!options?.allowPartialAddress && isPartialAddress(value)) {
```

Comment:
this `replacerFn` should be identical to the previous `stringifyFormReplacer`, with `options?.allowPartialAddress` being the only new part added.